examples/migration/recursive_fibonacci/task_emulation_layer.h

```diff
@@ -47,32 +47,49 @@ class base_task {
 public:
     base_task() = default;
 
-    base_task(const base_task& t) : m_parent(t.m_parent), m_child_counter(t.m_child_counter.load())
+    base_task(const base_task& t) : m_type(t.m_type), m_parent(t.m_parent), m_child_counter(t.m_child_counter.load())
     {}
 
     virtual ~base_task() = default;
 
     void operator() () const {
         base_task* parent_snapshot = m_parent;
-        const_cast<base_task*>(this)->execute();
-        if (m_parent && parent_snapshot == m_parent && m_child_counter == 0) {
-            if (m_parent->remove_reference() == 0) {
+        task_type type_snapshot = m_type;
+
+        base_task* bypass = const_cast<base_task*>(this)->execute();
+
+        bool is_task_recycled_as_child = parent_snapshot != m_parent;
+        bool is_task_recycled_as_continuation = type_snapshot != m_type;
+
+        if (m_parent && !is_task_recycled_as_child && !is_task_recycled_as_continuation) {
```

Comment:
After simplifying the bypass approach, I think you should be able to replace ``m_parent && !is_task_recycled_as_child && !is_task_recycled_as_continuation`` with just ``m_type != task_type::recycled``.